eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/annotation/load/VariantAnnotationLineMapper.java

```diff
@@ -0,0 +1,376 @@
+            // Otherwise just set SO terms
+        } else {
+            consequenceType.setSoTermsFromSoNames(Arrays.asList(lineFields[6].split(",")));   // fill so terms
+        }
+        currentAnnotation.getConsequenceTypes().add(consequenceType);
+        
+        return currentAnnotation;
+    }
+
+    /**
+     * From org.opencb.biodata.formats.annotation.io.VepFormatReader
+     * #parseRemainingFields(org.opencb.biodata.models.variant.annotation.ConsequenceType, java.lang.String[])
+     */
+    private void parseRemainingFields(ConsequenceType consequenceType, String[] lineFields) {
+        consequenceType.setEnsemblGeneId(lineFields[3]);    // fill Ensembl gene id
+        consequenceType.setEnsemblTranscriptId(lineFields[4]);  // fill Ensembl transcript id
+        if(!lineFields[7].equals("-")) {
+            consequenceType.setcDnaPosition(parseStringInterval(lineFields[7]));    // fill cdna position
+        }
+        if(!lineFields[8].equals("-")) {
+            consequenceType.setCdsPosition(parseStringInterval(lineFields[8]));  // fill cds position
+        }
+        if(!lineFields[9].equals("-")) {
+            consequenceType.setAaPosition(parseStringInterval(lineFields[9]));    // fill aa position
+        }
+        consequenceType.setAaChange(lineFields[10]);  // fill aa change
+        consequenceType.setCodon(lineFields[11]); // fill codon change
+        if(!lineFields[6].equals("") && !lineFields.equals("-")) {  // VEP may leave this field empty
+            consequenceType.setSoTermsFromSoNames(Arrays.asList(lineFields[6].split(",")));    // fill so terms
+        }
+    }
+
+    /**
+     * From org.opencb.biodata.formats.annotation.io.VepFormatReader#parseStringInterval(java.lang.String)
+     */
+    private Integer parseStringInterval(String stringInterval) {
+        String[] parts = stringInterval.split("-");
+        if(!parts[0].equals("?")) {
+            return Integer.valueOf(parts[0]);
+        } else if(parts.length>1 && !parts[1].equals("?"))  {
+            return Integer.valueOf(parts[1]);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * From org.opencb.biodata.formats.annotation.io.VepFormatReader#parseVariant(java.lang.String, java.lang.String)
+     */
+    private Map<String,String> parseVariant(String variantString, String coordinatesString) {
+//    private Map<String,String> parseVariant(String coordinatesString, String alternativeString) {
+
+        Map<String, String> parsedVariant = new HashMap<>(5);
+
+        try {
+            String[] variantLocationFields = coordinatesString.split("[:-]");
+//            parsedVariant.put("chromosome", variantLocationFields[0]);
+//            parsedVariant.put("start", variantLocationFields[1]);
+            parsedVariant.put("end", (variantLocationFields.length > 2) ? variantLocationFields[2] : variantLocationFields[1]);
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
Looks like the last version of the biodata library throws the exception instead of exiting, so don't bother back-porting it to older versions.